mw-webapp/src/dataAccessLogic/DayReportDAL.ts

```diff
@@ -121,4 +143,77 @@ export class DayReportDAL {
     await DayReportService.updateDayReportDTO(dayReportDTO, dayReport.uuid);
   }
 
+  /**
+   * Create student comment to DayReport
+   */
+  public static async createStudentComment(dayReportUuid: string) {
+    const oldDayReport = await DayReportDAL.getDayReport(dayReportUuid);
+
+    const updatedCell = [...oldDayReport.studentComments, unicodeSymbols.space];
+
+    const updatedDayReport: DayReport = {
+      ...oldDayReport,
+      studentComments: updatedCell,
+    };
+    await DayReportDAL.updateDayReport(updatedDayReport);
+  }
+
+  /**
+   * Update student comment to DayReport
+   */
+  public static async updateStudentComment(props: DayReportProps) {
+    const oldDayReport = await DayReportDAL.getDayReport(props.dayReportUuid);
+
+    const getUpdatedText = oldDayReport.studentComments.map((item: string, i: number) => {
+      if (i === props.index) {
+        return `${props.text}`;
+      }
+
+      return item;
+    });
+
+    const updatedDayReport: DayReport = {
+      ...oldDayReport,
+      studentComments: getUpdatedText,
+    };
+    await DayReportDAL.updateDayReport(updatedDayReport);
+  }
+
+  /**
+   * Create learned for today to DayReport
+   */
+  public static async createLearnedForToday(dayReportUuid: string) {
+    const oldDayReport = await DayReportDAL.getDayReport(dayReportUuid);
+
+    const updatedCell = [...oldDayReport.learnedForToday, unicodeSymbols.space];
+
+    const updatedDayReport: DayReport = {
+      ...oldDayReport,
+      learnedForToday: updatedCell,
+    };
+    await DayReportDAL.updateDayReport(updatedDayReport);
+  }
+
+  /**
+   * Update learnedForToday to DayReport
+   */
+  public static async updateLearnedForToday(props: DayReportProps) {
```

Comment:
why your DAL layer operate props? only UI layer responsible to work with props

Reply:
Deleted